WordPress/gutenberg
# Media & Text: Add caption support

## What?

Closes https://github.com/WordPress/gutenberg/issues/14604

## Why?

Caption support was missing for media in `Media & Text` block.

## How?

Added caption support for media in `Media & Text` block. 

## Testing Instructions
1. Add `Media & Text` block
2. Upload an image
3. Add a caption to the image
4. Save, Reload, check for caption under image in editor and front-end
5. Select `Use Featured Image` option, and verify image caption.

## Screenshots or screencast 
 
<img width="645" alt="image" src="https://github.com/user-attachments/assets/c28781d2-a5d9-44b1-b28e-9df132bb73a5">
